test/intl402/DurationFormat/prototype/format/invalid-arguments-throws.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.DurationFormat.prototype.format
+description: >
+  "format" basic tests for invalid arguments that should throw TypeError exception.
+info: |
+  Intl.DurationFormat.prototype.format(duration)
+  (...)
+  3. Let record be ? ToDurationRecord(duration)
+---*/
+
+const df = new Intl.DurationFormat();
+
+assert.throws(TypeError, () => { df.format(undefined) }, "undefined" );
+assert.throws(TypeError, () => { df.format(null) }, "null");
+assert.throws(TypeError, () => { df.format(true) }, "true");
+assert.throws(TypeError, () => { df.format(-12) }, "-12");
+assert.throws(TypeError, () => { df.format(-12n) }, "-12n");
+assert.throws(TypeError, () => { df.format(1) }, "1");
+assert.throws(TypeError, () => { df.format(2n) }, "2n");
+assert.throws(TypeError, () => { df.format({}) }, "plain object");
```

Comment:
Consider adding a test with an a property that isn't read (e.g. `{ year: 1 }`) and one where a supported property is explicitly set to undefined (e.g. `{ years: undefined }`).